types/cucumber/index.d.ts

```diff
@@ -23,9 +23,7 @@ export interface TableDefinition {
     hashes(): Array<{ [colName: string]: string }>;
 }
 
-export type StepDefinitionParam = string | number | CallbackStepDefinition | TableDefinition;
-
-export type StepDefinitionCode = (this: World, ...stepArgs: StepDefinitionParam[]) => PromiseLike<any> | any | void;
+export type StepDefinitionCode = (this: World, ...stepArgs: any[]) => PromiseLike<any> | any | void;
```

Comment:
See the final bullet point of [https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes](common mistakes). If this function could really return any, then the other two members of the union type aren't helping at all.

Reply:
I saw that, but as you can see this was already the case before. I didn't think it appropriate to fix that up in this pull request, since I'm targeting #19831. Besides, if that mistake was made here, it was probably made at other points in the file as well: it'd probably be best to fix them all at once.

Reply:
Seems like this is the only point in the file that a change like this is necessary, so it seems fine to roll it into this PR.